litert/vendors/qualcomm/debugger/optrace_profiling/run.py

```diff
@@ -0,0 +1,415 @@
+    # TODO(jiunkaiy): Set check=True after seg fault fix.
+    subprocess.run(
+        apply_pluin_main_cmd, check=False, cwd=Path(__file__).resolve().parent
+    )
+
+    extract_bytecode_lst = [
+        *bazel_run,
+        "//litert/tools:extract_bytecode",
+        "--",
+        f"--model_path={tmp_tflite_path}",
+        f"--output_dir={_ASSETS_DIR}",
+    ]
+    logging.debug(" ".join(str(item) for item in extract_bytecode_lst))
+    result = subprocess.run(
+        extract_bytecode_lst,
+        check=True,
+        cwd=Path(__file__).resolve().parent,
+        capture_output=True,
+        text=True,
+    )
+    os.remove(tmp_tflite_path)
+    logging.info("Subprocess output:\n%s", result.stderr)
+    # Parse the stderr to extract the path to the generated ctx.bin file.
+    # "Wrote ... bytes to '<context binary path>"
+    match = re.search(r"bytes to '([^']+)'", result.stderr)
+    output_path = None
+    if match:
+        output_path = Path(match.group(1))
+        logging.info("Extracted path:\n%s", output_path)
+    else:
+        raise RuntimeError("Failed to generate context binary.")
+    return output_path
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Given a ctx bin, generate QNN HTP Optrace Profiling."
+    )
+    parser.add_argument(
+        "--model", "-m", type=str, help="path to the tflite model", required=True
+    )
+    parser.add_argument(
+        "--output_dir", "-o", type=str, help="path to output folder", required=True
+    )
+    parser.add_argument(
+        "--hostname", "-H", type=str, help="hostname for adb", default="localhost"
+    )
+    parser.add_argument(
+        "--serial", "-s", type=str, help="serial for adb", required=True
+    )
+    parser.add_argument(
+        "--soc_model", type=str, help="SoC Model (e.g. SM8650)", required=True
+    )
+    parser.add_argument(
+        "--htp_arch", type=str, help="HTP Arch (e.g. V75)", required=True
+    )
+    parser.add_argument(
+        "--log_level",
+        default="INFO",
+        choices=["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"],
+        help="Set the logging level",
+    )
+    parser.add_argument(
+        "--qairt_sdk",
+        type=Path,
+        help="Path to qairt sdk folder",
+        required=False,
+        default=(
+            Path(os.environ["LITERT_QAIRT_SDK"]) / "latest"
+            if "LITERT_QAIRT_SDK" in os.environ
+            else Path(__file__).resolve().parents[5]
+            / "third_party"
+            / "qairt"
+            / "latest"
+        ),
+    )
+    args = parser.parse_args()
+    _setup_logging(args.log_level)
+    ctx_bin = _generate_ctx_bin(args.model, args.soc_model)
+    _generate_zero_inputs(_get_ctx_bin_info(ctx_bin, args.qairt_sdk))
+    ADB_CMD = _get_adb_cmd(args.hostname, args.serial)
+    _push_so(ADB_CMD, args.htp_arch, args.qairt_sdk)
+    _push_target(ADB_CMD, ctx_bin)
+    _run_ctx_bin(ADB_CMD, ctx_bin)
```

Comment:
I think we can merge these functions because they always work together and depend on the previous one.

```suggestion
    _run_ctx_bin_by_net_run_and_profile(ADB_CMD, args.htp_arch, args.qairt_sdk, ctx_bin)
```